src/coreclr/nativeaot/System.Private.CoreLib/src/System/RuntimeExceptionHelpers.cs

```diff
@@ -175,6 +175,27 @@ public static void RuntimeFailFast(RhFailFastReason reason, Exception? exception
             }
         }
 
+        internal const uint EXCEPTION_NONCONTINUABLE = 0x1;
+        internal const uint FAIL_FAST_GENERATE_EXCEPTION_ADDRESS = 0x1;
+        internal const uint STATUS_STACK_BUFFER_OVERRUN = 0xC0000409;
+        internal const uint FAST_FAIL_EXCEPTION_DOTNET_AOT = 0x48;
+
+#pragma warning disable 649
+        internal unsafe struct EXCEPTION_RECORD
+        {
+            internal uint ExceptionCode;
+            internal uint ExceptionFlags;
+            internal IntPtr ExceptionRecord;
+            internal IntPtr ExceptionAddress;
+            internal uint NumberParameters;
+#if TARGET_64BIT
+            internal fixed ulong ExceptionInformation[15];
+#else
+            internal fixed uint ExceptionInformation[15];
```

Comment:
is this a goo use case for nuint?

Reply:
C# only allows primitives with know size in this context. It would fail to compile with nuint.

If you wanted to avoid the ifdef, you can rewrite it as nuint inline array.

Reply:
What does an "inline" array look like? Yes, a fixed nuint E[15] doesn't compile.

Reply:
I'd rather not go through all that.  Probably can't index like an array like my code does.

Reply:
FWIW, it can be indexed. Roslyn generates indexer for the InlineArrays. Here is example of indexer being used on an inline array from a different PR: https://github.com/dotnet/runtime/pull/89519/files#diff-f4c84c289e159fe05ac08ce6846a73a7e1d0b00226da269c497cb7477185a58aR2633-R2634

It is fine to keep it as is with ifdef. I am just pointing out that converting this to InlineArray is viable way to avoid the ifdef. It won't simplify the code much though.

Reply:
Thanks.